scripts.js

```diff
@@ -0,0 +1,205 @@
+/*п. 1*/
+
+const loop = (times = 0, callback = null) => {
+    if (times > 0){
+        for (let i = 0; i < times; i++){
+            console.log(callback);
+        }
+    }
+}
+loop(3, 5); //задаем оба параметра
+loop(2); //оставляем callback по умолчанию
+loop(0, 2); //times = 0, в консоль не выводится
+ 
+
+
+
+/*п. 2*/
+
+const calculateArea = (figure, input = []) =>{
+    let area = null;
+    let curInput = [...input];
+    
+    switch (figure){
+        case('triangle'):
+            area = (curInput[0]*curInput[1]/2);    
+            break;   
+        case('parallelogram'):
+            area = (curInput[0]*curInput[1]);    
+            break;
+        case('circle'):
+            area = Math.pow((Math.PI*curInput[0]), 2);    
+            break;
+        default:
+            alert( 'Фигура неизвестна');   
+            
+    }
+    let result = {
+        area: area,
+        figure: figure,
+        input: curInput
+    }
+    console.log(result);
+}
+calculateArea('triangle', [4, 8]);
+calculateArea('parallelogram', [9, 7]);
+calculateArea('circle', [2]);
+
+
+
+
+/*п. 3*/
+
+const staff = {
+    managers: [],
+    developers: []
+}
+
+
+class Human {
+    constructor(name, dateOfBirth){
+        this.name = name;
+        this.dateOfBirth = new Date(dateOfBirth);
+        this.age = Math.floor((new Date() - this.dateOfBirth)/(365*24*3600*1000));
+    }
+    displayinfo(){
+        console.log(this.name + ', ' + this.age + ', ' + this.dateOfBirth);
+    }
+}
+
+class Employee extends Human {
+    constructor(name, dateOfBirth, salary, department){
+        super(name, dateOfBirth);
+        this.name = name;
+        this.salary = salary;
+        this.department = department;
+    }
+    
+    displayinfo(){
+        super.displayinfo();
+        console.log(this.salary + ', ' + this.department);
+    }
+}
+class Manager extends Employee {
+    constructor(name, dateOfBirth, salary, department, developers = []){
+        super(name, dateOfBirth, salary, department);
+        this.developers = developers;
+        this._removeDev();
+        this.addToStaff();
+    }
+    addToStaff(){
+        let manItem = {
+            name: this.name,
+            department: this.department,
+            developers: this.developers
+        }
+        staff.managers.push(manItem);
+    }
+    _removeDev(devname){
+        for (let developer in this.developers){
+            this.developers = this.developers.filter(developer => developer !== devname);
+            staff.developers = staff.developers.filter(developer => developer.name !== devname);
+            for (let managItem of [...staff.managers]){
+                if(this.developers.includes(devname)){
```

Comment:
```suggestion
                if(this.developers.includes(devname)){
```
этот `if` должен обводить внешний `for`